src/components/DataTable/components/Cell/Cell.tsx

```diff
@@ -99,6 +99,7 @@ function Cell({
 
   const headingMarkup = header ? (
     <th
+      data-polaris-header-cell
```

Comment:
You'll want to add an import in this component bringing in `headerCell`:

```
import {headerCell} from 'components/shared'
```

And then update this line to:

```suggestion
      {...headerCell.props}
```